### PR TITLE
feat: Update handling of tools_strict to ensure compatibility with vLLM

### PR DESCRIPTION
**Pull request by Emil-io**
### Proposed Changes:
Only add the "strict" dictionary to the openai_tools for generation, if tools_strict is available.

This ensures compatibility with vLLM (which adopts the OpenAI Interface). vLLM throws an error, if the "strict" dictionary is part of the tools list:

`openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': "[{'type': 'extra_forbidden', 'loc': ('body', 'tools', 0, 'function', 'strict'), 'msg': 'Extra inputs are not permitted', 'input': False}`

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack/blob/main/code_of_conduct.txt)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:` and added `!` in case the PR includes breaking changes.
- I documented my code
- I ran [pre-commit hooks](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md#installation) and fixed any issue

### CODE REVIEW

File: haystack_experimental/components/generators/chat/openai.py
Comment (julian-risch) [inline haystack_experimental/components/generators/chat/openai.py:452]:
```suggestion
                        **({"strict": tools_strict} if tools_strict is not None else {}),
```

Comment (julian-risch) [reply haystack_experimental/components/generators/chat/openai.py:452]: We should allow setting `tools_strict` to `false` and only set it to `{}` if no value was provided at all.

**Review by julian-risch — CHANGES_REQUESTED** [review]
I have a small change request but other than that the PR looks good to me. I would like to get another review from @vblagoje or @anakin87 before we merge.

**Review by anakin87 — APPROVED** [review]
LGTM!

**Review by vblagoje — APPROVED** [review]
LGTM as well

**Pull request merged**
